mcapi/mcapi.proto

```diff
@@ -0,0 +1,437 @@
+  Interval report_time = 2;
+
+  // Specification of for custom viewability and completion filter.
+  message CustomImpressionQualificationFilter {
+    // Specification for a quantile
+    enum Quantile {
+      QUANTILE_UNSPECIFIED = 0;
+      ZERO = 1;
+      TWENTY_FIVE = 2;
+      FIFTY = 3;
+      SEVENTY_FIVE = 4;
+      ONE_HUNDRED = 5;
+    }
+
+    // Specify a percent as a fraction or as a quantile
+    message Percent {
+      oneof value {
+        float fraction = 1;
+        Quantile quantile = 2;
+      }
+    }
+    Percent viewability = 1;
+    Percent completion = 2;
+  }
+
+  // Used to filter impressions by viewability and percent completion.
+  // If particular filter values are not applicable to the media they
+  // are applied to, the filters are ignored. (e.g. display cannot
+  // be filtered by percent completion)
+  message ImpressionQualificationFilter {
+    // The media type the filter applies to.
+    MediaType media_type = 1;
+
+    // Request that impressions be filtered by a configured
+    // filter type (e.g. AMI, MRC)
+    // The valid values and their configurations are specified as
+    // part of the market configuration.
+    repeated string filter_names = 2;
+
+    // A user defined filter.
+    CustomImpressionQualificationFilter custom = 3;
+  }
+
+  // The set of filters for this report. There is at most
+  // one filter per Media Type. Moreover, if any Reporting Set
+  // used in this Report is associated with any particular Media Type
+  // there must be a filter for that Media Type in this list.
+  // A filter for MediaType = CROSS_MEDIA must be present if
+  // any Pages are defined that contain Reporting Sets of
+  // heterogeneous Media Type.
+  //
+  // These filters are applied
+  // automatically based upon the Reporting Sets present in
+  // the PageDefinition selector. For example if all Reporting
+  // Sets in a page definition selector conform to the "video" media
+  // type then the "video" filter is applied. On the other hand
+  // if the media type is heterogenous then the cross-media
+  // filter is applied.
+  repeated ImpressionQualificationFilter qualification_filters = 3;
+
+  // Specifies how to Group output metrics.
+  message Grouping {
+    // The name of a report template field to group by.
+    // Only those event template fields that are tagged as groupable
+    // and in the common template are supported.
+    // Moreover, they must be enums (e.g. age bucket).
+    // This could be extended in the future to support non-enums and
+    // other grouping use cases..
+    repeated string dimension_name = 1;
+  }
+
+  // The set of additional filters to be applied to the page of results.
+  // Only those event templates that are specified in the "common" event
+  // template may be provided as filters. (e.g. age bucket)
+  message Filter {
+    message Term {
+      // The name of event template (aka dimension) to filter on
+      string dimension_name = 1;
```

Comment:
this is the name of the field in the event template, correct?

Reply:
Yes, it is the proto field name.